modal-api.md

```diff
@@ -0,0 +1,58 @@
+# Design ideas for modal dialogs
+
+## The simple approach
+
+Several of the UX modal designs are pretty simple, consisting of a title, some text and some buttons (examples). These prompt-like intefaces could be supported with a cross-frame message with the format:
+
+```TypeScript
+interface Prompt = {
+  title : string,
+  message : string,
+  button : array<Button>
+}
+
+interface Button = {
+  text : string,
+  type : 'secondary' | 'primary',
+  id : string
+}
+```
+
+When the host app displays the prompt and the user clicks a button, the client would be sent a message:
+
+```TypeScript
+interface ModalResult = {
+  buttonId : string
+}
+```
+
+With the caveat that buttonId can probably always be 'cancel', if the user closes the modal.
```

Comment:
Per my comment above, we'll might want to explicitly track the close method.  button, clickoutside, and close.  Generally, these are handled the same, but I've definitely had projects in the past where I wanted to handle these cases differently.